include a description of WORKFLOW_SHARE_DIR/bin and lib/python (#609)
cylc/cylc-doc: src/user-guide/writing-workflows/runtime.rst

The top level share directory location can be changed, e.g. to a large data
area, by global config settings under :cylc:conf:`global.cylc[install][symlink dirs]`.

If your workflow creates or installs executables or Python libraries
as it is running, these can be placed in:

* ``share/bin`` - for executables.
* ``share/lib/python`` - for Python modules.

> These folders will be added to ``PATH`` and ``PYTHONPATH`` so they can be
> used by jobs in the same way as

Reviewer: ```suggestion
These folders are automatically added to ``PATH`` and ``PYTHONPATH`` so they can be
used by jobs in the same way as the
```

> * ``share/bin`` - for executables.
> * ``share/lib/python`` - for Python modules.
> 
> These folders are automatically added to ``PATH`` and ``PYTHONPATH`` so they can be
> used by jobs in the same way as the
> :ref:`top level "bin/" and "lib/python/" directories <WorkflowDefinitionDirectories>`.
> 
> .. note::
> 
>    Cylc will not create these folders.
> 
> .. important::
> 
>    These locations take precedence over the top level ``bin/`` and
>    ``lib/python/`` directories in the run dir.
> 

Reviewer: ```suggestion
* ``share/bin/`` - for executables. This location is automatically added to ``PATH``
  (before the top-level ``bin/`` in the run dir).
* ``share/lib/python/`` - for Python modules. This location is automatically added
  to ``PYTHONPATH`` (before the top-level ``lib/python/`` in the run dir).

.. note::

   Cylc will not create these folders.
   
.. seealso::

   :ref:`Top level "bin/" and "lib/python/" directories <WorkflowDefinitionDirectories>`.

```

>   These locations will be placed first in the executable and Python module
>   search paths, before those in the parent run directory .

Reviewer: Maybe
```suggestion
   These locations take precedence over the top level ``bin/`` and
   ``lib/python/`` directories in the run dir.
```

Reviewer 2: @wxtim and @MetRonnie - this kind of reverse my 2nd review suggestion above.  I would prefer to just "state the facts" here, which is where these directories are put in the search paths.  "Take precedence" or "called first" (Tim's original phrasing) could be assumed to have wider implications, e.g. for how Cylc itself somehow manages these locations.

Author: I rather agree with Hilary - can we come to a conclusion @MetRonnie ?

Reviewer: https://github.com/cylc/cylc-doc/pull/609/files#r1211855271

The top level share directory location can be changed, e.g. to a large data
area, by global config settings under :cylc:conf:`global.cylc[install][symlink dirs]`.

> If your workflow creates or installs executables or python libraries

Reviewer 2: ```suggestion
If your workflow creates or installs executables or Python libraries
```

as it is running, these can be placed in:

* ``share/bin`` - for executables.
* ``share/lib/python`` - for Python modules.

These folders will be added to ``PATH`` and ``PYTHONPATH`` so they can be
used by jobs in the same way as
:ref:`top level "bin/" and "lib/python/" directories <WorkflowDefinitionDirectories>`.

.. note::

   Cylc will not create these folders.

.. important::

>    Scripts placed in these directories will be called instead of a script
>    of the same name in ``bin`` or ``lib/python`` directories at the top
>    level of the workflow.

Reviewer 2: I would just state the facts here, in case users over-interpret that statement. Something like this:

```suggestion
  These locations will be placed first in the executable and Python module
  search paths, before those in the parent run directory .
```
